### ISSUE TITLE
running JVB as root

### ISSUE THREAD

**Issue by roeycohen** [jitsi/docker-jitsi-meet#100]
Hi, 

i'd like to run JVB on port 80 (<= 1024), for that i need to JVB to run as root.
Any ideas on where to start?

also, will let's encrypt work if HTTP_PORT is set to a port different than 80?

10x!

**Comment by roeycohen** [comment]
it's you should change jvb to root in this line, can someone on confirm?
https://github.com/jitsi/docker-jitsi-meet/blob/e1bd2447ae4f5e90e0acea9bf95e23b1960f70e9/jvb/rootfs/etc/services.d/jvb/run#L15

**Comment by sapkra** [comment]
You should NEVER run containers as root for security reasons.

**Comment by saghul** [comment]
I think enabling some of the capabilities (CAP_NET_ADMIN maybe?) you can bind low ports as a user.

As for LetsEncrypt, no, it only works on port 80/443.

**Comment by roeycohen** [comment]
10q guys for replying! :)

@sapkra - why not? it's the only service in it's container, and other services in the jitsi stack are running as root (e.g. nginx).

@saghul - i'll take a look on it. do you have experience with using port 80 instead 10,000? do you think it will work?

**Comment by sapkra** [comment]
@roeycohen There are a lot of articles about it in the web. Here is the first one I've found.

https://medium.com/@mccode/processes-in-containers-should-not-run-as-root-2feae3f0df3b

Yeah I think all containers are running as root right now but in my environment I can not start containers with root because this is restricted. To get them running I forked the repository and fixed the permissions so that the containers are working without root.
It's still on my list to cleanup my changes and create an PR to follow the recommended security standard.

**Comment by roeycohen** [comment]
@sapkra 10x!

running jbv on lower ports seems to fail...
```
JVB 2019-05-19 08:51:03.384 WARNING: [2163] org.jitsi.videobridge.EndpointMessageTransport.log() SCTP connection with drdsdocx not ready yet.
JVB 2019-05-19 08:51:03.385 WARNING: [2163] org.jitsi.videobridge.EndpointMessageTransport.log() No available transport channel, can't send a messag
```

guess it wasn't meant to be 😪

**Comment by saghul** [comment]
> It's still on my list to cleanup my changes and create an PR to follow the recommended security standard in the official images.

❤️

**Comment by saghul** [comment]
> running jbv on lower ports seems to fail...

Have you modified the container to run privileged or passed in the cap I mentioned?

**Comment by roeycohen** [comment]
> Have you modified the container to run privileged or passed in the cap I mentioned?
no, just ran the container as root (i guess it should be enough for testing...)

**Comment by roeycohen** [comment]
made some more tests, indeed, if port <= 1023 - no data is sent to jitsi.

**Comment by saghul** [comment]
I’m not sure that’s enough.

**Comment by roeycohen** [comment]
@saghul are you referring to CAP_NET_ADMIN? because the JVB process isn't running as a service but by calling /init command (at least for this dockers project)

**Comment by saghul** [comment]
yes, that's the one. Hum, good point, /init calls it... I wonder how to solve this. The webserver does bind to 80 and 443 so I guess it should just work 🤷‍♂

**Comment by roeycohen** [comment]
i believe it does, because i get different error messages...

**Comment by saghul** [comment]
Can you paste a full error log? Those lines above in this thread don't seem related, I think.

**Comment by roeycohen** [comment]
Hi Saghul,
attaching you two logs, one with 10,000 port which is working and the other is 80 which isn't.
[80_notworking.log](https://github.com/jitsi/docker-jitsi-meet/files/3206539/80_notworking.log)
[10000_working.log](https://github.com/jitsi/docker-jitsi-meet/files/3206540/10000_working.log)

appreciate your time :)

**Comment by saghul** [comment]
I don't see any failures in the log. Is port 80 UDP open in the firewall?

**Comment by roeycohen** [comment]
don't think there's a firewall involved... 
the dockers run on VirtualBox installed on my machine and also added an outbound rule in windows firewall to allow UDP 80.
maybe you can learn something from the chrome log?
[jitsi.blossom.office-1558531392019.log](https://github.com/jitsi/docker-jitsi-meet/files/3207541/jitsi.blossom.office-1558531392019.log)
